File: python/CTDL_002-TongDayConBangK.py
n = int(input())
k = int(input())
a = n*[0]
dem = 0
for i in range(0,n):
    a.append(int(input()))

Bin = n*[0]
def process():
    sum = 0
    for i in range(0,n):
        if Bin[i] == 1:
           sum += a[i]
    if sum == k:
        dem +=1
        for i in range(0,n):
            if Bin[i] == 1:
                print(a[i],end=' ')
def In():
    for i in Bin:
        print(i,end=' ')   
    print('\n')
def Try(i):
    for j in range(0,2):
        Bin[i] = j
        if i==n-1:
            process()
        else:
            Try(i+1)

Try(0)

File: C++/CTDL_002-TongDayConBangK.cpp
#include<bits/stdc++.h>
using namespace std;
int n,k,bin[100],dem = 0,a[100];
void process(){
    int sum = 0;
    for(int i=1;i<=n;i++)
        if(bin[i])
            sum+=a[i];
    if(sum==k){
        dem ++;
        for(int i=1;i<=n;i++)
            if(bin[i])
                cout << a[i]<<" ";
        cout << "\n";
    }
}
void Try(int i){
    for(int j=0;j<2;j++){
        bin[i]=j;
        if(i==n)
            process();
        else
            Try(i+1);
    }
}
int main(){
    cin >> n >> k;
    for(int i=1;i<=n;i++)
        cin >> a[i];
    Try(1);
    cout << dem;
}